compose/config/config_schema_v4.0.json

```diff
@@ -1,10 +1,7 @@
 {
   "$schema": "http://json-schema.org/draft-04/schema#",
-  "id": "config_schema_v3.8.json",
+  "id": "config_schema_v4.0.json",
```

Comment:
compose-spec.json

Reply:
in an ideal world, this whole file should be removed from docker/compose repo and replaced by a (build-time?) reference to https://github.com/compose-spec/compose-spec/blob/master/schema/config_schema.json

Reply:
did you checked if the current status of the compose-spec schema already includes all those changes or if there's anything missing ?

Reply:
I started from the compose-spec schema and added properties from v2.x and v3.x.